client/signup/steps/user/index.jsx

```diff
@@ -22,6 +22,19 @@ export default React.createClass( {
 		}
 	},
 
+	componentWillMount() {
```

Comment:
Actually, you should do the calculation in `componentWillReceiveProps` AND `componentWillMount` because `componentWillReceiveProps` is not called for the initial render. To that end the calculation itself should probably be its own method.